Update the main CLI page with changes for 0.12 (#2782)
temporalio/documentation: docs/cli/index.mdx

  - term
---

The Temporal CLI provides direct access to a Temporal Service via the terminal.
It's a powerful tool for managing, monitoring, and debugging Temporal Applications.
You can use it to start, stop, inspect and operate on Workflows and Activities, and perform administrative tasks such as Namespace, Schedule, and Task Queue management.

> The Temporal CLI also includes an embedded Temporal Cluster suitable for use in development and CI/CD.

Reviewer: ```suggestion
The Temporal CLI also includes an embedded Temporal Service suitable for use in development and CI/CD.
```

We are in the process of updating "Temporal Cluster" to "Temporal Service".

Register Namespaces with `namespace create`:

```shell

> temporal operator namespace create -n foo

Reviewer 2: ```suggestion
temporal operator namespace create --namespace foo

temporal operator namespace create -n foo
```

As a tech writing rule, prefer the full "readable" version to the shortcut. Clarity trumps concision. 

Several ways to approach this, use the full version, use both with the preferred first, or stick with organizational standard (-n) and ignore suggestion. This is a due diligence suggestion. 

The extra characters take up more space, especially in help messages, but I was scratching my head a lot when I was first exposed to `-n` in messages and had no context to explain it. As a new adopter being shown --namespace first and then knowing as a Unix user there might be a shorter flag, would have made some work (coverage of creating a namespace using the Temporal CLI) easier for me. I expect other new adopters, a big target I've been told for approachable tooling and documentation, would also benefit.

Reviewer 2: Also, leaving this here because I don't know how to add it to the points where it applies (starting on line 25). The intro can be simplified for easier reading.

Starting at Line 25 -> 

The Temporal CLI, formerly 'tctl', is a command-line interface that offers direct access to a Temporal Cluster (Service).
It supports management, monitoring, and debugging of Temporal Applications, and administrative functions like Namespace management. 
With it, you can initiate, signal, or cancel Workflows and query Workflow states from your terminal.
The Temporal CLI is an indispensable component of your Temporal developer toolkit.

When upgrading from `tctl` to Temporal CLI, update your existing environment variables and be aware that some commands use new syntax.

| `TEMPORAL_ADDRESS`                       | Host and port (formatted as host:port) for the Temporal Frontend Service. | --address                       |

> | `TEMPORAL_CODEC_AUTH`                    | Sets the authorization header for requests to Codec Server.               | --codec-auth                    |

Reviewer 2: All the others start with the important noun, not a a description of use. To be consistent:
```suggestion
| `TEMPORAL_CODEC_AUTH`                    | Authorization header for requests to Codec Server.               | --codec-auth                    |
```

### zsh auto-completion

> 1. Add the following line to your `~/.zshrc` file:

Reviewer 2: The pedant in me:

```suggestion
1. Add the following line to your `~/.zshrc` configuration file:
```

Author: This is may be a little pedantic in turn, but it's actually a script, not a configuration file—it can run arbitrary code when `zsh` is launched. The `zsh` manpage calls it a "shell startup file", the `bash` manpage calls it an "initalization file`; I think I'll go with "startup script" if that works for you?

Reviewer 2: If memory serves, rc means "run commands" so have at it. Generally, I have described it as a configuration file that customizes the launch of a shell. Adding some kind of modifier explains what you're changing to any people who don't know _why_ they're editing the file. I like shell initialization slightly more than just initialization.

   ```sh
   eval "$(temporal completion zsh)"
   ```

2. Re-launch your shell or run:

   ```sh
   source ~/.zshrc
   ```

### Bash auto-completion

1. Install [bash-completion](https://github.com/scop/bash-completion#installation) and add the software to your `~/.bashrc`.

2. Add the following line to your `~/.bashrc` file:

   ```sh

>    eval "$(temporal completion bash)"

Reviewer 2: Comment: tested this with the default Apple Bash install, and found it errored during completion. I leave my system as "vanilla" as possible, and as close to the default installs that people will encounter when first using their computer (in this case macOS Sonoma, Bash).

```
bash-3.2$ temporal serbash: _get_comp_words_by_ref: command not found
ver
```

Author: Do you have `bash-completion` installed? This error usually means the completion library isn't loaded in your shell. I can add a note on that in the docs.

Author: 👍 I tested it too! And apparently had bash-completion installed even though I use zsh, heh. I blame Homebrew. :)

   ```

3. Re-launch your shell or run:

   ```sh
   source ~/.bashrc
   ```

### Fish auto-completion

> 1. Make sure there is a place to store custom completions for Fish:

Reviewer 2: A little more clarity that is action-oriented rather than "checking"

```suggestion
1. Create the Fish custom completions if one does not already exist:
```

   ```fish
   mkdir -p ~/.config/fish/completions
   ```

> 2. Arrange for the completions to be loaded when needed (note: the file name must be `temporal.fish` or the completions will not be found):

Reviewer 2: ```suggestion
2. Configure the completions to load when needed. Note: the file name must be `temporal.fish` or the completions will not be found:
```
